lms/product/product.py

```diff
@@ -0,0 +1,61 @@
+"""Core models of the product."""
+
+from dataclasses import asdict, dataclass
+from enum import Enum
+
+from lms.services.grouping.plugin import GroupingServicePlugin
+
+
+class Family(str, Enum):
+    """Enum for which product this relates to."""
+
+    BLACKBAUD = "BlackbaudK12"
+    BLACKBOARD = "BlackboardLearn"
+    CANVAS = "canvas"
+    D2L = "desire2learn"
+    MOODLE = "moodle"
+    SAKAI = "sakai"
+    SCHOOLOGY = "schoology"
+    UNKNOWN = "unknown"
+
+    @classmethod
+    def _missing_(cls, _value):
+        return cls.UNKNOWN
+
+
+@dataclass
+class Plugins:
+    """A collection of plugins used to separate LMS specific functionality."""
+
+    grouping_service: GroupingServicePlugin
+
+
+@dataclass
+class PluginConfig:
+    """A collection of plugin class definitions."""
+
+    # These also provide the default implementations
+    grouping_service: type = GroupingServicePlugin
```

Comment:
I might have inlined this directly in `Product`, mostly because we only have one now.  I see  the appeal of having it in a separated container, probably more future proof.

Reply:
I think it makes it easier to see the structure of the product without loads of indentation